### PR TITLE
Create a theme model that inherits from a base class with serialization information.

### PR DESCRIPTION
**Pull request by OlgaLarina**

### CODE REVIEW

File: packages/survey-creator-core/src/components/tabs/header-model.ts [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+    getVerticalAlignment("logoPositionY", "top"),
+    getHorizontalAlignment("titlePositionX", getLocString("theme.coverTitlePosition"), "left"),
+    getVerticalAlignment("titlePositionY", "bottom"),
+    getHorizontalAlignment("descriptionPositionX", getLocString("theme.coverDescriptionPosition"), "left"),
+    getVerticalAlignment("descriptionPositionY", "bottom"),
+  ]);
+
+Serializer.addProperties("headersettings", [
+  {
+    type: "fontsettings",
```
Comment (RomanTsukanov) [inline packages/survey-creator-core/src/components/tabs/header-model.ts:388]:
```suggestion
    type: "font",
```

File: packages/survey-creator-core/src/components/tabs/header-model.ts [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+    visibleIf: (obj) => obj.headerView === "basic",
+    default: getDefaultTitleSetting(),
+    onSerializeValue: (obj: HeaderModel) => {
+      return onSerializeFontSettingsValue(obj, "surveyTitle");
+    }
+  },
+  {
+    type: "fontsettings",
+    name: "surveyDescription",
+    displayName: getLocString("theme.surveyDescription"),
```
Comment (RomanTsukanov) [inline packages/survey-creator-core/src/components/tabs/header-model.ts:400]:
```suggestion
    name: "surveyDescriptionFont",
    displayName: getLocString("theme.surveyDescriptionFont"),
```

File: packages/survey-creator-core/src/components/tabs/header-model.ts [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+    visibleIf: (obj) => obj.headerView === "basic",
+    default: getDefaultDescriptionSetting(),
+    onSerializeValue: (obj: HeaderModel) => {
+      return onSerializeFontSettingsValue(obj, "surveyDescription");
+    }
+  },
+  {
+    type: "fontsettings",
+    name: "headerTitle",
+    displayName: getLocString("theme.surveyTitle"),
```
Comment (RomanTsukanov) [inline packages/survey-creator-core/src/components/tabs/header-model.ts:410]:
```suggestion
    name: "headerTitleFont",
    displayName: getLocString("theme.surveyTitleFont"),
```

File: packages/survey-creator-core/src/components/tabs/theme-custom-questions/element-settings.ts [hunk 4]
```diff
@@ -21,23 +22,17 @@ function getElementsJSON() {
     },
     {
       type: "spinedit",
-      name: "corner",
```
Comment (RomanTsukanov) [inline packages/survey-creator-core/src/components/tabs/theme-custom-questions/element-settings.ts:24]: I don't like `"colorsettings"` above. Can we rename it to simply `"color"`?

File: packages/survey-creator-core/src/components/tabs/theme-model.ts [hunk 5]
```diff
@@ -0,0 +1,1173 @@
...
+      if (!!editor) {
+        editor.visibleIf = "{advancedmode} = true";
+        editor.descriptionLocation = "hidden";
+      }
+    },
+    onSerializeValue: (obj: ThemeModel) => {
+      return onSerializeFontSettingsValue(obj, "pageDescription");
+    }
+  }, {
+    type: "boxshadowsettings",
```
Comment (RomanTsukanov) [inline packages/survey-creator-core/src/components/tabs/theme-model.ts:931]:
```suggestion
    type: "boxshadow",
```

File: packages/survey-creator-core/src/components/tabs/theme-model.ts [hunk 6]
```diff
@@ -0,0 +1,1173 @@
...
+      if (!!editor) {
+        editor.unit = "%";
+        editor.min = 0;
+        editor.max = 100;
+        editor.step = 5;
+        editor.titleLocation = "left";
+      }
+    }
+  }, {
+    type: "headersettings",
```
Comment (RomanTsukanov) [inline packages/survey-creator-core/src/components/tabs/theme-model.ts:837]:
```suggestion
    type: "header",
```

File: packages/survey-creator-core/src/components/tabs/theme-model.ts [hunk 7]
```diff
@@ -0,0 +1,1173 @@
...
+        if (!!editor) {
+          editor.visibleIf = "{advancedmode} = true";
+          editor.descriptionLocation = "hidden";
+        }
+      },
+      onSerializeValue: (obj: ThemeModel) => {
+        return obj.onSerializeElementSettingsValue("editorPanel");
+      }
+    }, {
+      type: "elementsettings",
```
Comment (RomanTsukanov) [inline packages/survey-creator-core/src/components/tabs/theme-model.ts:770]: What's `"elementsettings"`?

File: packages/survey-creator-core/src/components/tabs/theme-custom-questions/element-settings.ts (hunk 4 above)
Comment (OlgaLarina) [reply packages/survey-creator-core/src/components/tabs/theme-custom-questions/element-settings.ts:24]: The color editor already exists. This editor is a combined color and transparency editor. We need to come up with something else.

**Review by tsv2013 — APPROVED** [review]

**Pull request merged**